Fix misspells in the code base

We're about to enable Go linter checks on CI, as a part of that, I would like to see all issues inside current codebase fixed (at least ones that are easy to fix).

This issue is about warnings found by `go misspell` linter.

```
src/langsrv/langsrv.go:339:16: `convertion` is a misspelling of `conversion` (misspell)
// very simple convertion
               ^
src/linter/custom.go:24:5: `Preferrably` is a misspelling of `Preferably` (misspell)
	// Preferrably something unique, prefixed with a vendor
	   ^
src/linter/root.go:924:20: `mispelled` is a misspelling of `misspelled` (misspell)
	// Check commonly mispelled types and other unfortunate cases.
	                  ^
src/linter/root.go:933:11: `happend` is a misspelling of `happened` (misspell)
		// This happend when either of those formats is used:
		        ^
src/linttest/basic_test.go:894:49: `overriden` is a misspelling of `overridden` (misspell)
	// variables declared using @var should not be overriden
	                                               ^
```

I would like to see all the typos to be fixed in 1 PR, if possible.

--- a/src/langsrv/langsrv.go
+++ b/src/langsrv/langsrv.go
@@ -336,7 +336,7 @@ func handleTextDocumentSymbol(req *baseRequest) error {
 	return nil
 }
 
-// very simple convertion
+// very simple conversion
 func posToLocation(pos meta.ElementPosition) vscode.Location {
 	return vscode.Location{
 		URI: "file://" + pos.Filename,

--- a/src/linter/custom.go
+++ b/src/linter/custom.go
@@ -21,7 +21,7 @@ type MetaCacher interface {
 	// When underlying cache structure is updated, version
 	// should return different value.
 	//
-	// Preferrably something unique, prefixed with a vendor
+	// Preferably something unique, prefixed with a vendor
 	// name, like `mylints-1.0.0` or `extension-abc4`.
 	//
 	// Returned value is written before Encode() is called to

--- a/src/linter/root.go
+++ b/src/linter/root.go
@@ -921,7 +921,7 @@ func (d *RootWalker) checkPHPDocType(typ string) string {
 		return fmt.Sprintf("%s type syntax: use [] after the type, e.g. T[]", typ)
 	}
 
-	// Check commonly mispelled types and other unfortunate cases.
+	// Check commonly misspelled types and other unfortunate cases.
 	switch typ {
 	case "boolean":
 		return "use bool type instead of boolean"
@@ -930,7 +930,7 @@ func (d *RootWalker) checkPHPDocType(typ string) string {
 	case "long", "integer":
 		return "use int type instead of " + typ
 	case "-":
-		// This happend when either of those formats is used:
+		// This happened when either of those formats is used:
 		// `* @param $name - description`
 		// `* @param - $name description`
 		// We don't want to make "-" slip as a type name.

--- a/src/linttest/basic_test.go
+++ b/src/linttest/basic_test.go
@@ -891,7 +891,7 @@ func TestUnused(t *testing.T) {
 }
 
 func TestAtVar(t *testing.T) {
-	// variables declared using @var should not be overriden
+	// variables declared using @var should not be overridden
 	_ = linttest.GetFileReports(t, `<?php
 	function test() {
 		/** @var string $a */

